trace/tr_test.go

```diff
@@ -34,23 +34,56 @@ func TestTrace(t *testing.T) {
 		return
 	}
 
-	ctx := context.Background()
-	var spanReg *trace.Registrar
-	defer trace.StartServer(&ctx, 1, &spanReg)()
+	var ctx context.Context
+
+	_ = lager.ExitViaPanic()
+
+	os.Unsetenv("GCP_PROJECT_ID")
+	ctx = nil
+	ex := u.GetPanic(func(){
+		StartServer(&ctx, 1)
+	})
+	u.IsNot(nil, ex, "Start fails w/o proj")
+	u.Like(logs.Bytes(), "Start no proj logs",
+		"[cC]ould not start Registrar")
+	logs.Reset()
+	os.Setenv("GCP_PROJECT_ID", proj)
+
+	os.Setenv("GOOGLE_APPLICATION_CREDENTIALS", "/no-creds")
```

Comment:
Probably OOS for this PR, but have you considered the application default credentials flow which contains some other bits and bobs? https://google.aip.dev/auth/4110

Reply:
I thought I was using that or much of it.  This is why the integration works with just the setting of GCP_PROJECT_ID.  No extra work to provide creds is done when running tests on my laptop.  And no extra work to get creds is required when code runs in GKE.  And the determination of project ID happens automatically in GKE.

The code you commented on is setting up fake creds to trigger failures to get slightly more code coverage.

But I'll look over your link some more later.  Thanks!

Reply:
No problem! Yeah my thinking was in the situation where you might need to request an OAuth2 access token and use it directly without going through a client library/env-vars, if I recall correctly. I may have missed it in the code, so just wanted to ask if that was possible with this implementation! :) Here is the link the developer docs: https://cloud.google.com/compute/docs/access/create-enable-service-accounts-for-instances#applications